src/types/index.ts

```diff
@@ -15,8 +15,8 @@ export enum Network {
 
 export interface TokenDetails {
   name?: string
-  symbol?: string
-  decimals?: number
+  symbol: string
```

Comment:
Why is symbol required? 
I would understand `decimals` if we want to assume `18` as the default.

Also, in another change, you added `Required` to `TokenDetails` and then only extracted the `symbol`, so also not coherent with this change.